### PR TITLE
Bump gnuradio4 and adopt API

### PR DESCRIPTION
**Pull request by wirew0rm**
GNURadio 4.0 has seen some breaking API changes, most notably the new reflection API, changes to the Buffer/Reader/Span API and some minor renamings.
This commit bumps GR4 to the latest commit and follows up all breaking changes.

There is one issue remaining which has to be resolved before merging, probably related to https://github.com/fair-acc/gnuradio4/pull/441
Basically the processBulk signature for array of port changed (while vector of port stayed the same) and now the process work of picoscope would be expected to have a separate parameter for each port.

Also adds restyled automatic formatting.

### CODE REVIEW

File: blocklib/picoscope/Picoscope4000a.hpp [hunk 1]
```diff
@@ -12,9 +12,11 @@ class Picoscope4000a : public fair::picoscope::Picoscope<T, acquisitionMode, Pic
 public:
     using super_t = fair::picoscope::Picoscope<T, acquisitionMode, Picoscope4000a<T, acquisitionMode>>;
 
-    Picoscope4000a(gr::property_map props) : super_t(std::move(props)) {}
+    Picoscope4000a(gr::property_map props) : super_t(std::move(props)) {
+        analog_out.resize(8);
+    }
 
-    std::array<gr::PortOut<T>, 8> analog_out;
+    std::vector<gr::PortOut<T>> analog_out{};
```
Comment (frankosterfeld) [inline blocklib/picoscope/Picoscope4000a.hpp:19]: `analog_out{8}` should also work, no? Using `explicit vector( size_type count,
                 const Allocator& alloc = Allocator() );`

File: blocklib/picoscope/Picoscope5000a.hpp [hunk 2]
```diff
@@ -12,9 +12,13 @@ class Picoscope5000a : public fair::picoscope::Picoscope<T, acquisitionMode, Pic
 public:
     using super_t = fair::picoscope::Picoscope<T, acquisitionMode, Picoscope5000a<T, acquisitionMode>>;
 
-    Picoscope5000a(gr::property_map props) : super_t(std::move(props)) {}
+    Picoscope5000a(gr::property_map props) : super_t(std::move(props)) {
+        analog_out.resize(4);
+    }
+
+    std::vector<gr::PortOut<T>> analog_out{};
```
Comment (frankosterfeld) [inline blocklib/picoscope/Picoscope5000a.hpp:19]: `analog_out{8}`?

File: CMakeLists.txt [hunk 3]
```diff
@@ -27,134 +29,157 @@ include(FetchContent)
 
 set(ENABLE_TESTING OFF)
 FetchContent_Declare(
-    graph-prototype
-    GIT_REPOSITORY https://github.com/fair-acc/graph-prototype.git
-    GIT_TAG 4b05d3059335d14e9fa2eb44b43bb37a4881409c # main as of 2024-06-18
+  gnuradio4
+  GIT_REPOSITORY https://github.com/fair-acc/gnuradio4.git
+  GIT_TAG 582a759020b310ba2e4b47efd5c0f2984d886489 # main as of 2024-10-11
```
Comment (frankosterfeld) [inline CMakeLists.txt:34]: Can you bump this even more to the now-latest main?

**Review by drslebedev — APPROVED** [review]
Thanks a lot for the PR!

Changes look good to me, PR can be merged!

**Pull request merged**